src/lib/container/smartlist.c

```diff
@@ -188,6 +188,31 @@ smartlist_ints_eq(const smartlist_t *sl1, const smartlist_t *sl2)
   return 1;
 }
 
+/**
+ * Return true if there is shallow equality between smartlists -
+ * i.e. all indices correspond to exactly same object (pointer
+ * values are matching). Otherwise, return false.
+ */
+int
+smartlist_shallow_eq(const smartlist_t *s1, const smartlist_t *s2)
+{
+  if (s1 == NULL || s2 == NULL)
```

Comment:
if `s1 == NULL && s2 == NULL` I think this function should return 1.

Reply:
Fixed in ac40d0e22ed097fb5cca5eb31b7d308ffd545947.